src/drift.d.ts

```diff
@@ -0,0 +1,302 @@
+// Drift API types
+declare class Drift {
+  invoked: boolean
+  SNIPPET_VERSION: string
+  api: DriftAPI
+  allFramesReady: boolean
+  apiReady: boolean
+  controllerReady: boolean
+  hasInitialized: boolean
+  chatReady: boolean
+  debug(): void
+  config(config: DriftWidgetConfig): void
+  init(embedId: string): void
+  load(widgetId: string): void
+  unload(): void
+  hide(): void
+  show(): void
+  reset(): void
+  page(name: string): void
+  identify(userId: string, attributes: DriftUser): void
+  track(event: string, attributes: Record<string, unknown>[]): void
+  on(
+    event: 'ready',
+    callback: (
+      api: DriftAPI,
+      payload: {
+        isOnline: boolean
+        sidebarOpen: boolean
+        teamAvailability: Record<string, unknown>[]
+      },
+    ) => void,
+  ): void
+}
+
+type DriftWidgetConfig = {
+  locale?: string
+  messages?: {
+    welcomeMessage?: string
+    awayMessage?: string
+    emailCaptureMessage?: string
+    thankYouMessage?: string
+  }
+  enableWelcomeMessage?: boolean
+  enableCampaigns?: boolean
+  enableChatTargeting?: boolean
+  backgroundColor?: string
+  foregroundColor?: string
+  autoAssignee?: {
+    name: string
+    title: string
+    email: string
+    avatarUrl?: string
+  }
+  inboxId?: number
+  cookieDomain?: string
+  disableNewConversations?: boolean
+}
+
+type DriftUser = {
+  email?: string
+  nickname?: string
+  age?: number
+}
+
+declare class DriftAPI {
+  goToConversation(conversationId: number): void
+  goToConversationList(): void
+  goToNewConversation(message: string): void
+  hideAwayMessage(): void
+  hideChat(): void
+  hideWelcomeMessage(): void
+  openChat(): void
+  scheduleMeeting(driftUserId: number | string): void
+  setUserAttributes(attributes: DriftUser): void
+  showAwayMessage(): void
+  /**
+   * @deprecated The method should not be used
+   * https://devdocs.drift.com/v1.1/docs/welcome-message
+   */
+  showWelcomeMessage(message: string): void
+  showWelcomeOrAwayMessage(): void
+  sidebar: {
+    open(): void
+    close(): void
+    toggle(): void
+  }
+  startInteraction(interaction: {
+    interactionId: number | string
+    goToConversation: boolean
+  }): void
+  startVideoGreeting(): void
+  toggleChat(): void
+  widget: {
+    hide(): void
+    show(): void
+  }
+  // https://devdocs.drift.com/docs/drift-events#email-captured
+  on(
+    event: 'emailCapture',
+    callback: (payload: {
+      email: string
+      conversationId: number
+      playbookId?: number
+      interactionId?: number
+      campaignId?: number
+    }) => void,
+  ): void
```

Comment:
seems like all of those following triggers need to be on the DriftObject not on the API...

Reply:
nope, that is not correct. Drift sets them both on the drift and on the API object, however, they will only be set on the drift object after drift is ready. Since you need wait for drift before adding callbacks and drift returns API, it only really makes sense to add them to the API object. See #4 for example usage